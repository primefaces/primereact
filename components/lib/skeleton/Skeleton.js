import * as React from 'react';
import { classNames, ObjectUtils } from '../utils/Utils';

export const Skeleton = React.memo(React.forwardRef((props, ref) => {
    const otherProps = ObjectUtils.findDiffKeys(props, Skeleton.defaultProps);
    const style = props.size ?
        { width: props.size, height: props.size, borderRadius: props.borderRadius } :
        { width: props.width, height: props.height, borderRadius: props.borderRadius };
    const className = classNames('p-skeleton p-component', {
        'p-skeleton-circle': props.shape === 'circle',
        'p-skeleton-none': props.animation === 'none'
    }, props.className);

    return <div style={style} className={className} {...otherProps}></div>
}));

Skeleton.displayName = 'Skeleton';
Skeleton.defaultProps = {
    __TYPE: 'Skeleton',
    shape: 'rectangle',
    size: null,
    width: '100%',
    height: '1rem',
    borderRadius: null,
    animation: 'wave',
    style: null,
    className: null
}
