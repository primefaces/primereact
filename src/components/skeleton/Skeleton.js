import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { classNames } from '../utils/ClassNames';

export class Skeleton extends Component {

    static defaultProps = {
        shape: 'rectangle',
        size: null,
        width: '100%',
        height: '1rem',
        borderRadius: null,
        animation: 'wave',
        style: null,
        className: null
    }

    static propTypes = {
        shape: PropTypes.string,
        size: PropTypes.string,
        width: PropTypes.string,
        height: PropTypes.string,
        borderRadius: PropTypes.string,
        animation: PropTypes.string,
        style: PropTypes.object,
        className: PropTypes.string
    };

    skeletonStyle() {
        if (this.props.size)
            return { width: this.props.size, height: this.props.size, borderRadius: this.props.borderRadius };
        else
            return { width: this.props.width, height: this.props.height, borderRadius: this.props.borderRadius };
    }

    render() {
        const skeletonClassName = classNames('p-skeleton p-component', {
            'p-skeleton-circle': this.props.shape === 'circle',
            'p-skeleton-none': this.props.animation === 'none'
        }, this.props.className);

        const style = this.skeletonStyle();

        return (
            <div style={style} className={skeletonClassName}></div>
        );
    }
}
