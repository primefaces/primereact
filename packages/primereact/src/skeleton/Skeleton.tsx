'use client';
import { Component, withComponent } from '@primereact/core/component';
import { useSkeleton } from '@primereact/headless/skeleton';
import { styles } from '@primereact/styles/skeleton';
import { mergeProps } from '@primeuix/utils';
import * as React from 'react';
import { defaultProps } from './Skeleton.props';

export const Skeleton = withComponent({
    defaultProps,
    styles,
    setup: (instance) => {
        const skeleton = useSkeleton(instance.inProps);

        return skeleton;
    },
    render: (instance) => {
        const {
            id,
            props,
            ptmi,
            cx,
            sx,
            // element refs
            elementRef
        } = instance;

        const style = props.size ? { width: props.size, height: props.size, borderRadius: props.borderRadius } : { width: props.width, height: props.height, borderRadius: props.borderRadius };

        const rootProps = mergeProps(
            {
                id,
                className: cx('root'),
                style: { ...style, ...sx('root') },
                'aria-hidden': true
            },
            ptmi('root')
        );

        return <Component as={props.as || 'div'} {...rootProps} ref={elementRef} />;
    }
});
