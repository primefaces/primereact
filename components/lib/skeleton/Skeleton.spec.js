import '@testing-library/jest-dom';
import { Skeleton } from './Skeleton';

import { snapshot } from '../../test';

describe('Skeleton', () => {
    snapshot(<Skeleton />, 'default');
    snapshot(<Skeleton shape="circle" />, 'shape circle');
    snapshot(<Skeleton shape="rectangle" />, 'shape rectangle');
    snapshot(<Skeleton animation="none" />, 'animation none');
    snapshot(<Skeleton animation="wave" />, 'animation wave');
    snapshot(<Skeleton borderRadius="16px" />, 'border radius');
    snapshot(<Skeleton borderRadius="16px" />, 'border radius');
    snapshot(<Skeleton width="100%" height="2rem" />, 'size width and height');
    snapshot(<Skeleton size="50px" />, 'size in pixels');
});
