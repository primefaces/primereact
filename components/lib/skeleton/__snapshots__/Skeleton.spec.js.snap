// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton animation none 1`] = `
<div>
  <div
    class="p-skeleton p-component p-skeleton-none"
    style="width: 100%; height: 1rem;"
  />
</div>
`;

exports[`Skeleton animation wave 1`] = `
<div>
  <div
    class="p-skeleton p-component"
    style="width: 100%; height: 1rem;"
  />
</div>
`;

exports[`Skeleton border radius 1`] = `
<div>
  <div
    class="p-skeleton p-component"
    style="width: 100%; height: 1rem; border-radius: 16px;"
  />
</div>
`;

exports[`Skeleton border radius 2`] = `
<div>
  <div
    class="p-skeleton p-component"
    style="width: 100%; height: 1rem; border-radius: 16px;"
  />
</div>
`;

exports[`Skeleton default 1`] = `
<div>
  <div
    class="p-skeleton p-component"
    style="width: 100%; height: 1rem;"
  />
</div>
`;

exports[`Skeleton shape circle 1`] = `
<div>
  <div
    class="p-skeleton p-component p-skeleton-circle"
    style="width: 100%; height: 1rem;"
  />
</div>
`;

exports[`Skeleton shape rectangle 1`] = `
<div>
  <div
    class="p-skeleton p-component"
    style="width: 100%; height: 1rem;"
  />
</div>
`;

exports[`Skeleton size in pixels 1`] = `
<div>
  <div
    class="p-skeleton p-component"
    style="width: 50px; height: 50px;"
  />
</div>
`;

exports[`Skeleton size width and height 1`] = `
<div>
  <div
    class="p-skeleton p-component"
    style="width: 100%; height: 2rem;"
  />
</div>
`;
